### ISSUE TITLE
Duplicated traceId's when using Baggages

### ISSUE THREAD

**Issue by aswarcewicz** [micrometer-metrics/tracing#572]
Hi!
I've been observing duplicated traceId's when using `.contextWrite(ReactorBaggage.append(...))`
It looks like everything is fine when contextWrite is used once. Something wrong is happen when used twice.
Please checkout this repo from branch baggage-tests: https://github.com/aswarcewicz/spring-demo-otel/tree/baggage-tests and open localhost:8080. I'm sure that some of the requests will have the same traceId

**Comment by marcingrzejszczak** [comment]
Can you elaborate on "Something wrong is happen when used twice." ?

**Comment by aswarcewicz** [comment]
I mean that I don’t know where exactly source of bug is, but I have observed that ‘contextWrite’ can be used only once, when used second time it causes duplicates of traceid

**Comment by marcingrzejszczak** [comment]
@chemicL any ideas why is that happening?

**Comment by aswarcewicz** [comment]
Do you need additional details from me?

**Comment by marcingrzejszczak** [comment]
Is this still a problem? I upgraded to tracing 1.2.6 and have sent a few requests and the trace id is different each time...
